src/test/groovy/cz/smarteon/loxone/app/LoxoneAppTest.groovy

```diff
@@ -16,28 +18,88 @@ class LoxoneAppTest extends Specification implements SerializationSupport {
         then:
         config.lastModified == LAST_MODIFIED
         config.miniserverInfo?.name == 'ShowRoom'
+        config.rooms.size() == 3
+        config.rooms.values().first() instanceof Room
+        config.categories.size() == 3
+        config.categories.values().first() instanceof Category
         config.controls.size() == 6
         config.controls.values().first() instanceof AlarmControl
     }
 
     def "should getControl by type"() {
         given:
-        LoxoneApp config = new LoxoneApp(LAST_MODIFIED, Mock(MiniserverInfo), [(UUID): alarmControl])
+        LoxoneApp config = new LoxoneApp(
+                LAST_MODIFIED,
+                Mock(MiniserverInfo),
+                [(UUID_ROOM): room],
+                [(UUID_CATEGORY): category],
+                [(UUID_CONTROL): alarmControl])
 
         expect:
         config.getControl(AlarmControl) == alarmControl
 
         where:
         alarmControl << [new AlarmControl(), Stub(AlarmControl)]
+        room << [new Room(), Stub(Room)]
+        category << [new Category(), Stub(Category)]
     }
 
     def "should getControl by name and type"() {
         given:
         def control = new SwitchControl()
         control.name = 'SomeControl'
-        LoxoneApp config = new LoxoneApp(LAST_MODIFIED, Mock(MiniserverInfo), [(UUID): control])
+        def room = new Room()
+        room.name = "SomeRoom"
+        def category = new Category()
+        category.name = "SomeCategory"
+        LoxoneApp config = new LoxoneApp(
+                LAST_MODIFIED,
+                Mock(MiniserverInfo),
+                [(UUID_ROOM): room],
+                [(UUID_CATEGORY): category],
+                [(UUID_CONTROL): control])
 
         expect:
         config.getControl('SomeControl', SwitchControl) == control
     }
+
+    def "should getControls by room"() {
+        when:
+        LoxoneApp config = readResource('app/LoxAPP3.json', LoxoneApp)
+        LoxoneUuid roomUuid = new LoxoneUuid("0f869a64-028d-0cc2-ffffd4c75dbaf53c")
+
+        then:
+        def room = config.getRooms().get(roomUuid)
+        room != null
```

Comment:
could be inlined...

Reply:
Not entirely clear what you mean, i'm not a groovy expert ;-)

Reply:
the variable `room` is used only once, so it must not be decared at all, the code could be just like:
`config.getRooms().get(roomUuid) != null`

Let me know if you are willing to change it - I don't insist on it

Reply:
Unless i'm sleeping ;-) i use the variable on line (at the moment)

Reply:
ahh my bad...